### PR TITLE
Feature popular

### PR DESCRIPTION
**Pull request by MillanUka**
- Added the Popular Items feature
- No-longer use Hard-coded values in LoadCatalog and ItemManager
- Minor Bug fixes
- Reconnected the Preview Buttons with the Manipulation Scene
- Changes to Database Input and Output

### CODE REVIEW

File: Assets/scripts/Item.cs [hunk 1]
```diff
@@ -140,4 +140,24 @@ public Boolean Equals(Object obj)
...
+        }
+        else if(numberOfClicks < otherItemClicks)
+        {
+            return 1;
+        }
+        else
+        {
+            return 0;
+        }
+    }
```
Comment (MinhPhu0304) [inline Assets/scripts/Item.cs]: You could just do things like ```return  numberOfClicks - otherItemClick;``` and the result is the same

File: Assets/scripts/ItemManager.cs [hunk 2]
```diff
@@ -149,17 +148,41 @@ private void LoadDBRecordFrom(IDataReader dbRecord)
             double price = dbRecord.GetFloat(2);
             string itemSiteURL = dbRecord.GetString(3);
             string itemDesc = dbRecord.GetString(4);
+            Debug.Log(dbRecord.GetString(5));
```
Comment (MinhPhu0304) [inline Assets/scripts/ItemManager.cs]: No debug log please

File: Assets/scripts/ItemManager.cs [hunk 3]
```diff
@@ -184,17 +211,37 @@ private void AddDemoDataToDB()
...
+
+            SqliteParameter P_update_name = new SqliteParameter("@noClick", item.getNumberOfClick());
+            SqliteParameter P_update_id = new SqliteParameter("@id", item.GetItemID());
+
+            dbcmd.Parameters.Add(P_update_name);
+            dbcmd.Parameters.Add(P_update_id);
+
+            dbcmd.CommandText = sqlQuery;
+            dbcmd.ExecuteScalar();
+            dbconn.Close();
```
Comment (MinhPhu0304) [inline Assets/scripts/ItemManager.cs]: The dbconn will close by its self. I think we have to do call ```dbcmd.close();``` to close the command

**Review by MinhPhu0304 — CHANGES_REQUESTED** [review]

File: Assets/scripts/ItemManager.cs (hunk 3 above)
Comment (MillanUka) [reply Assets/scripts/ItemManager.cs]: There is no dbcmd.close(),

**Review by MinhPhu0304 — APPROVED** [review]

**Pull request merged**